go/codegen/experimental/xgboost.go

```diff
@@ -153,14 +166,55 @@ def step_entry():
         dtrain = xgb_dataset(ds, train_fn, select, feature_metas,
                              feature_column_names, label_meta, is_pai,
                              pai_train_table, transform_fn=transform_fn)
-        dval = xgb_dataset(ds, val_fn, val_select, feature_metas,
-                           feature_column_names, label_meta, is_pai,
-                           pai_train_table, transform_fn=transform_fn)
+        if val_select:
+            dval = xgb_dataset(ds, val_fn, val_select, feature_metas,
+                               feature_column_names, label_meta, is_pai,
+                               pai_train_table, transform_fn=transform_fn)
+        else:
+            dval = None
         eval_result = runtime.{{.Submitter}}.xgboost.train(dtrain, train_params, model_params, dval)
 `
 
+func generateFeatureColumnCode(fcList []ir.FeatureColumn) (string, error) {
+	fcCodes := make([]string, 0, len(fcList))
+	for _, fc := range fcList {
+		// xgboost have no cross feature column, just get the first field desc
+		fd := fc.GetFieldDesc()[0]
+		tmpl := `fc.%s(fd.FieldDesc(name="%s", dtype=fd.DataType.%s, delimiter="%s", format=fd.DataFormat.%s, shape=%s, is_sparse=%s, vocabulary=%s))`
+		fcTypeName := reflect.TypeOf(fc).Elem().Name()
+		isSparseStr := "False"
+		if fd.IsSparse {
+			isSparseStr = "True"
+		}
+		vocabList := []string{}
+		for k := range fd.Vocabulary {
+			vocabList = append(vocabList, k)
+		}
+		dataFormat := "PLAIN"
+		if fd.Format != "" {
+			dataFormat = fd.Format
```

Comment:
In the Go side, `Format` is in lower case, but in the Python side, `dataFormat` is in upper case.